consensus/qbft/src/test/java/org/hyperledger/besu/consensus/qbft/validator/ForkingValidatorProviderTest.java

```diff
@@ -141,7 +141,8 @@ public void migratesFromContractToBlockValidatorProvider() {
     assertThat(validatorProvider.getValidatorsForBlock(genesisHeader))
         .isEqualTo(CONTRACT_ADDRESSES_1);
     assertThat(validatorProvider.getValidatorsForBlock(header1)).isEqualTo(CONTRACT_ADDRESSES_1);
-    assertThat(validatorProvider.getValidatorsForBlock(header2)).isEqualTo(BLOCK_ADDRESSES);
+    // should use previous validatorProvider's validators
+    assertThat(validatorProvider.getValidatorsForBlock(header2)).isEqualTo(CONTRACT_ADDRESSES_1);
```

Comment:
TODO not sure if this is semantically correct - the removal of `block == forkSpec.getBlock()` as pointed out in the other comment is responsible for the difference